Adding typescript

rafinskipg opened the pull request:
- Add tsdx to compile build items, this generates cjs , esm and umd modules (including types)
- Change alll package.jsons to include types, build types, and test commands
- Remove babel configuration
- Remove webpack

BREAKING CHANGES:
- multicall schemas `getCdps` and `userVaultsList`: formerly optional `descending` argument is now *required*. This is because of how typescript transpiles functions with optional arguments. A check in `MulticallService.ts` uses `function.length` to check the expected number of arguments and throws an error if it doesn't match with the actual arguments. Without typescript, arguments with default values aren't counted, however with typescript, they are included. See more: https://medium.com/@jishaal/what-is-function-length-ece130546fee

rafinskipg committed:
Compiles, but does not work for tests

rafinskipg committed:
First tests for dai core

rafinskipg committed:
Lint

rafinskipg committed:
In process of going green

rafinskipg committed:
test: trying to get test to work

rafinskipg committed:
ci

rafinskipg committed:
ci

rafinskipg committed:
target es5

rafinskipg committed:
timeouts

rafinskipg committed:
Migrate to tsdx for compiling to multiple targets, migrated module dai

rafinskipg committed:
services core and liquidations

rafinskipg committed:
dai-plugin-scd

rafinskipg committed:
dai-plugin-mcd

rafinskipg committed:
dai-plugin-mcd

rafinskipg committed:
dai-plugin-governance

rafinskipg committed:
fix: Tests for governance plugin

rafinskipg committed:
fix dependencies and build

rafinskipg committed:
Cross fingers

rafinskipg committed:
more fixes

rafinskipg committed:
Cross fingers

rafinskipg committed:
Cross fingers

rafinskipg committed:
Cross fingers

rafinskipg committed:
trying to get test to run

rafinskipg committed:
trying to get test to run

rafinskipg committed:
fix servicess core

rafinskipg committed:
trying to fix services core dependencies

rafinskipg committed:
Kill me

rafinskipg committed:
more stuff

rafinskipg committed:
Lint

rafinskipg committed:
Merge branch 'dev' of github.com:makerdao/dai.js into typescript

rafinskipg committed:
Trying to move forward with this

rafinskipg committed:
lint

rafinskipg committed:
Fix build

b-pmcg committed:
fix multicall schema tests

rafinskipg committed:
UMD bundle works

rafinskipg committed:
Merged with dev

rafinskipg committed:
lint

rafinskipg committed:
Publish

 - @makerdao/dai-plugin-governance@0.15.0-alpha.0
 - @makerdao/dai-plugin-liquidations@0.2.0-alpha.0
 - @makerdao/dai-plugin-mcd@1.7.0-alpha.0
 - @makerdao/dai-plugin-migrations@1.4.0-alpha.0
 - @makerdao/dai-plugin-scd@1.3.0-alpha.0
 - @makerdao/dai@0.41.0-alpha.0
 - @makerdao/services-core@0.11.0-alpha.0
 - @makerdao/test-helpers@0.4.0-alpha.0

rafinskipg committed:
modified packages

rafinskipg committed:
Updatge maker

rafinskipg committed:
Lint

rafinskipg committed:
Publish

 - @makerdao/dai-plugin-governance@0.15.0-alpha.1
 - @makerdao/dai-plugin-liquidations@0.2.0-alpha.1
 - @makerdao/dai-plugin-mcd@1.7.0-alpha.1
 - @makerdao/dai-plugin-migrations@1.4.0-alpha.1
 - @makerdao/dai-plugin-scd@1.3.0-alpha.1
 - @makerdao/dai@0.41.0-alpha.1
 - @makerdao/services-core@0.11.0-alpha.1
 - @makerdao/test-helpers@0.4.0-alpha.1

rafinskipg committed:
Merged with dev

rafinskipg committed:
Publish

 - @makerdao/dai-plugin-governance@0.15.0-alpha.2
 - @makerdao/dai@0.41.0-alpha.2

rafinskipg committed:
update peer deps

rafinskipg committed:
Publish

 - @makerdao/dai-plugin-governance@0.15.0-alpha.3
 - @makerdao/dai-plugin-liquidations@0.2.0-alpha.2
 - @makerdao/dai-plugin-mcd@1.7.0-alpha.2
 - @makerdao/dai-plugin-migrations@1.4.0-alpha.2
 - @makerdao/dai-plugin-scd@1.3.0-alpha.2
 - @makerdao/dai@0.41.0-alpha.3
 - @makerdao/services-core@0.11.0-alpha.2
 - @makerdao/test-helpers@0.4.0-alpha.2

rafinskipg committed:
versions

rafinskipg committed:
Merge branch 'dev' of github.com:makerdao/dai.js into typescript

rafinskipg committed:
remove presets copyfiles

rafinskipg committed:
large

b-pmcg reviewed (approved):
![](https://media.giphy.com/media/KIxBn0rcMbhII/giphy.gif)

rafinskipg committed:
Update deps

rafinskipg committed:
Update deps

rafinskipg committed:
patch package based on  https://github.com/facebook/jest/issues/9983#issuecomment-936055633

rafinskipg committed:
prepatch

rafinskipg committed:
Merged with dev

rafinskipg committed:
remove ethereum utils

rafinskipg committed:
lint

rafinskipg committed:
Publish

 - @makerdao/dai-plugin-governance@0.15.0-alpha.4
 - @makerdao/dai-plugin-liquidations@0.2.0-alpha.3
 - @makerdao/dai-plugin-mcd@1.7.0-alpha.3
 - @makerdao/dai-plugin-migrations@1.4.0-alpha.3
 - @makerdao/dai-plugin-scd@1.3.0-alpha.3
 - @makerdao/dai@0.41.0-alpha.4
 - @makerdao/services-core@0.11.0-alpha.3
 - @makerdao/test-helpers@0.4.0-alpha.3

rafinskipg committed:
version
